### PR TITLE
[testing] Fix e2e tests timed out, but there's empty test log

### PR DESCRIPTION
**Pull request by Bobgy**
Recent e2e tests usually fail with
```
Argo workflow timed out
```

e.g. in PR: https://github.com/kubeflow/pipelines/pull/2603

~~It seems 30 minutes limit is no longer enough.~~ I will investigate while time increases.
EDIT: something hangs the `go test` command, but log is not streamed out. So we should first make test log stream during testing.

/priority p0
/area testing
/kind bug

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/2810)

### CODE REVIEW

**Comment by Bobgy** [comment]
api integration tests seem to be the one that doesn't finish: https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubeflow_pipelines/2603/kubeflow-pipeline-e2e-test/1214467257921441792/artifacts/pods_info/integration-test-h7lf7-3930211570.txt
in https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/kubeflow_pipelines/2603/kubeflow-pipeline-e2e-test/1214467257921441792

**Comment by Bobgy** [comment]
Looked into the issue together with @jingzhang36, log for the running test is strange:
https://console.cloud.google.com/logs/viewer?project=ml-pipeline-test&advancedFilter=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22ml-pipeline-test%22%0Aresource.labels.location%3D%22us-east1-b%22%0Aresource.labels.cluster_name%3D%22e2e-ce0de86-26641%22%0Aresource.labels.namespace_name%3D%22kubeflow%22%0Aresource.labels.pod_name%3D%22integration-test-h7lf7-3930211570%22

It is almost empty:
```
go test -v ./... -namespace kubeflow -args -runIntegrationTests=true
```

Comparing with a succeeded test: https://console.cloud.google.com/logs/viewer?project=ml-pipeline-test&advancedFilter=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22ml-pipeline-test%22%0Aresource.labels.location%3D%22us-east1-b%22%0Aresource.labels.cluster_name%3D%22e2e-ce0de86-26641%22%0Aresource.labels.namespace_name%3D%22kubeflow%22%0Aresource.labels.pod_name%3D%22integration-test-h7lf7-3930211570%22

It seems the problem is: https://github.com/kubeflow/pipelines/blob/07efee5ee2aaf0fbd676a627527496b924a9cd6e/test/api-integration-test/run_test.sh#L67

The test command doesn't print anything if it hangs somewhere.
We need to first fix this issue to see what went wrong in the test.

**Comment by Bobgy** [comment]
Error in cloudbuild:
```
Step #1 - "scheduledworkflow": go: gopkg.in/tomb.v1@v1.0.0-20141024135613-dd632973f1e7: unrecognized import path "gopkg.in/tomb.v1" (parse https://gopkg.in/tomb.v1?go-get=1: no go-import meta tags ())
```

Example 2
```
Step #0 - "persistenceagent": go: gopkg.in/airbrake/gobrake.v2@v2.0.9: unrecognized import path "gopkg.in/airbrake/gobrake.v2" (parse https://gopkg.in/airbrake/gobrake.v2?go-get=1: no go-import meta tags ())
```

It seems the problem is gopkg website having higher network failure rate now than usual. Any network failure will fail go module requirements from finishing.

**Comment by Bobgy** [comment]
https://github.com/golang/go/issues/28194 should be the best way to reduce flakiness caused by unstable import websites, but there's been no progress on that issue much.

We can add some retry to hopefully mitigate flakiness in our presubmit tests.

**Comment by Bobgy** [comment]
/assign @Ark-kun 
Can you help me review?

**Comment by Bobgy** [comment]
Take a look at succeeded pod log, this fix is correct: https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/kubeflow_pipelines/2810/kubeflow-pipeline-e2e-test/1214503372829233152/artifacts/pods_info/integration-test-mlpb7-1035667546.txt

**Comment by Bobgy** [comment]
/assign @jingzhang36 
Can you help me review too?

File: test/initialization-test/run_test.sh [hunk 1]
```diff
@@ -64,13 +64,12 @@ cd "${BASE_DIR}/${TEST_DIR}"
...
-TEST_EXIT_CODE=$?
+LOG_FILE=$(mktemp)
+TEST_EXIT_CODE=0 # reference for how to save exit code: https://stackoverflow.com/a/18622662
+go test -v ./... -namespace ${NAMESPACE} -args -runIntegrationTests=true |& tee $LOG_FILE || TEST_EXIT_CODE=$?
 
-# Log the test result
-printf '%s\n' "$TEST_RESULT"
 # Convert test result to junit.xml
-printf '%s\n' "$TEST_RESULT" | go-junit-report > ${JUNIT_TEST_RESULT}
+cat $LOG_FILE | go-junit-report > ${JUNIT_TEST_RESULT}
```
Comment (Ark-kun) [inline test/initialization-test/run_test.sh]: Nit:
```suggestion
< "$LOG_FILE" go-junit-report > "${JUNIT_TEST_RESULT}"
```

File: test/api-integration-test/run_test.sh [hunk 2]
```diff
@@ -64,13 +64,12 @@ cd "${BASE_DIR}/${TEST_DIR}"
 export GO111MODULE=on
 
 echo "Run integration test..."
-TEST_RESULT=`go test -v ./... -namespace ${NAMESPACE} -args -runIntegrationTests=true 2>&1`
-TEST_EXIT_CODE=$?
+LOG_FILE=$(mktemp)
+TEST_EXIT_CODE=0 # reference for how to save exit code: https://stackoverflow.com/a/18622662
+go test -v ./... -namespace ${NAMESPACE} -args -runIntegrationTests=true |& tee $LOG_FILE || TEST_EXIT_CODE=$?
```
Comment (Ark-kun) [inline test/api-integration-test/run_test.sh]: Are you sure `$?` will be the exit code of `go` and not `tee`?

Comment (Bobgy) [reply test/api-integration-test/run_test.sh]: Thanks for the catch! You are right. tee affects exit code.
Added `set -o pipefail`

**Pull request merged**